diff --git a/compiler.mjs b/compiler.mjs
index 5dbc4a784b955e5e0f82fd87aa2b5b0fe86d6900..ad629e11f895c9f1d1dd7e3169a007dcf5e1a87f 100644
--- a/compiler.mjs
+++ b/compiler.mjs
@@ -39417,12 +39417,12 @@ function ssr(component, options) {
         ...reactive_store_subscriptions,
         instance_javascript,
         ...parent_bindings,
-        css.code && b `$$result.css.add(#css);`,
+        css.code && options.css === "injected" && b `$$result.css.add(#css);`,
         main
     ].filter(Boolean);
     const css_sourcemap_enabled = check_enable_sourcemap(options.enableSourcemap, 'css');
     const js = b `
-		${css.code ? b `
+		${css.code && options.css === "injected" ? b `
 		const #css = {
 			code: "${css.code}",
 			map: ${css_sourcemap_enabled && css.map ? string_literal(css.map.toString()) : 'null'}