diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 753e960b6d6078ebeb0cbfb0e9e03e526dc6fd36..0000000000000000000000000000000000000000
diff --git a/compiler.mjs b/compiler.mjs
index 9e78052cd043a9874ba5547bb6e42fbbb80a7c2f..fa425bf9568b9046a58e346322ccce7fc70f6a14 100644
--- a/compiler.mjs
+++ b/compiler.mjs
@@ -38478,7 +38478,7 @@ function dom(component, options) {
 				constructor(options) {
 					super();
 
-					${css.code && b `
+					${css.code && options.css === "injected" && b `
 						const style = document.createElement('style');
 						style.textContent = \`${css.code.replace(regex_backslashes, '\\\\')}${css_sourcemap_enabled && options.dev ? `\n/*# sourceMappingURL=${css.map.toUrl()} */` : ''}\`
 						this.shadowRoot.appendChild(style)`}
@@ -39417,7 +39417,7 @@ function ssr(component, options) {
         ...reactive_store_subscriptions,
         instance_javascript,
         ...parent_bindings,
-        css.code && b `$$result.css.add(#css);`,
+        css.code && options.css === "injected" && b `$$result.css.add(#css);`,
         main
     ].filter(Boolean);
     const css_sourcemap_enabled = check_enable_sourcemap(options.enableSourcemap, 'css');