diff --git a/src/compiler/compile/render_ssr/index.js b/src/compiler/compile/render_ssr/index.js
index 7a48c7f7e38f3df0ba7e34c07dddea5e811337de..cc4133217703a4f04947e838e148f0b1998214e3 100644
--- a/src/compiler/compile/render_ssr/index.js
+++ b/src/compiler/compile/render_ssr/index.js
@@ -175,13 +175,13 @@ export default function ssr(component, options) {
 		...reactive_store_subscriptions,
 		instance_javascript,
 		...parent_bindings,
-		css.code && b`$$result.css.add(#css);`,
+		css.code && options.css === "injected" && b`$$result.css.add(#css);`,
 		main
 	].filter(Boolean);
 	const css_sourcemap_enabled = check_enable_sourcemap(options.enableSourcemap, 'css');
 	const js = b`
 		${
-			css.code
+			css.code && options.css === "injected"
 				? b`
 		const #css = {
 			code: "${css.code}",
